blockchain/common_test.go

```diff
@@ -151,12 +152,12 @@ func newFakeChain(params *chaincfg.Params) *BlockChain {
 		deploymentCaches:              newThresholdCaches(params),
 		index:                         index,
 		bestChain:                     newChainView(node),
+		recentBlocks:                  lru.NewKVCache(1000),
```

Comment:
Why set this to 1000 in these tests rather than the actual `recentBlockCacheSize` value being used?

Reply:
Some of the synthetic treasury tests exercise code that needs to fetch blocks from the database (or cache), but the blocks used in the synthetic tests are mocked out and thus can't actually go in the database, so they cheat by putting them in the cache.  If the cache is too small, the blocks get evicted and the tests fail.

That said, now that I think about it, it would be better to just override the cache for each specific test in question and set it to the exact number the tests needs versus doing it for every mocked fake chain by default with a magic number.

Updated.